Makefile

```diff
@@ -40,51 +40,131 @@ buildnocache:
+		-e AWS_ACCESS_KEY_ID=${OLD_AWS_ACCESS_KEY_ID} \
+		-e AWS_SECRET_ACCESS_KEY=${OLD_AWS_SECRET_ACCESS_KEY} \
+		-e PGPASSWORD=postgres \
+		--entrypoint python \
+		--user=$(CURRENT_UID) \
+		$(API_SERVICE) \
+		manage.py dbbackup $(target)
+else
+	@echo "Please specify a target. \nie: make dbbackup target=local"
+endif
+
+db:
+	@echo "Test"
 
 dbrestore:
-	@docker-compose exec --user=$(CURRENT_UID) $(API_SERVICE) python manage.py dbrestore local
+ifdef target
+	@docker-compose run \
+		--rm \
+		--name api_dbrestore \
+		-e AWS_ACCESS_KEY_ID=${OLD_AWS_ACCESS_KEY_ID} \
+		-e AWS_SECRET_ACCESS_KEY=${OLD_AWS_SECRET_ACCESS_KEY} \
+		-e PGPASSWORD=postgres \
+		--entrypoint python \
+		--user=$(CURRENT_UID) \
+		$(API_SERVICE) \
+		manage.py dbrestore $(target)
+else
+	@echo "Please specify a target. \nie: make dbrestore target=local"
+endif
 
 migrate:
-	@docker-compose exec --user=$(CURRENT_UID) $(API_SERVICE) python manage.py migrate
-
+	@docker-compose run \
+		--rm \
+		--name api_migrate \
+		--entrypoint python \
+		--user=$(CURRENT_UID) \
+		$(API_SERVICE) \
+		manage.py migrate
+# -----------------
+# Setup actions
+# -----------------
 install:
 	@echo "\n--- Shutting down existing stack ---\n"
 	@make down
 	@echo "\n--- Building new docker image ---\n"
-	@make buildnocache
-	@make up
-	@echo "\n--- Spinning up new stack ---\n"
-	@sleep 20
-	@echo "\n--- Applying MERMAID database migrations ---\n"
+	@make build
+	@echo "\n--- Migrate ---\n"
 	@make migrate
-
+	@echo "\n--- Spinning up new stack ---\n"
+	@make up
 
 freshinstall:
 	@echo "\n--- Shutting down existing stack ---\n"
 	@make downnocache
 	@echo "\n--- Building new docker image ---\n"
 	@make buildnocache
-	@make up
-	@echo "\n--- Spinning up new stack ---\n"
-	@sleep 20
 	@echo "\n--- Restoring MERMAID database ---\n"
 	@make dbrestore
-	@echo "\n--- Applying MERMAID database migrations ---\n"
+	@echo "\n--- Migrate ---\n"
 	@make migrate
-
+	@echo "\n--- Spinning up new stack ---\n"
+	@make up
+	
+# override the default entrypoint and use runserver (for development)
 runserver:
-	@docker-compose exec --user=$(CURRENT_UID) $(API_SERVICE) python manage.py runserver 0.0.0.0:8080
+	@docker-compose run \
+		--rm \
+		--name api_runserver \
+		--workdir /code \
+		--volume $(PWD)/src:/code \
+		--publish 8080:8080 \
+		--entrypoint python \
+		--user=$(CURRENT_UID) \
+		$(API_SERVICE) \
+		manage.py runserver 0.0.0.0:8080
 
 shell:
-	@docker-compose exec --user=$(CURRENT_UID) $(API_SERVICE) /bin/bash
+	@docker exec \
+		--user=root \
+		-it \
+		api_runserver \
+		bash
 
 test:
-	@docker-compose exec --user=$(CURRENT_UID) $(API_SERVICE) pytest -v --no-migrations --rich api/tests
\ No newline at end of file
+	@docker-compose run \
+		--rm \
+		--name api_test \
+		-e AWS_ACCESS_KEY_ID=${OLD_AWS_ACCESS_KEY_ID} \
+		-e AWS_SECRET_ACCESS_KEY=${OLD_AWS_SECRET_ACCESS_KEY} \
+		--entrypoint pytest \
+		--user=$(CURRENT_UID) \
+		$(API_SERVICE) \
+		-v --no-migrations api/tests
```

Comment:
Is this right? `--no-migrations` is a flag for `pytest`; can that come between the `-v` flag and its arg?

Reply:
entrypoint is `pytest`, line 159 is the args that go with it